components/src/dynamo/planner/monitoring/perf_metrics.py

```
@@ -0,0 +1,252 @@
+
+        if isinstance(benchmark_data, str):
+            benchmark_data = json.loads(benchmark_data)
+
+        if isinstance(benchmark_data, dict) and benchmark_data.get("status") == "error":
+            logger.info(
+                f"get_perf_metrics returned error: {benchmark_data.get('message')}"
+            )
+            return []
+
+        return _extract_fpms_from_benchmark(benchmark_data, component_type)
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)